### PR TITLE
feat: add extended hardware information to Server and ServerClass CRDs

### PR DESCRIPTION
**Pull request by smira**
This is final part of #735, previous part in #823.

This imports all remaining changes with some fixups minus the webhook
changes.

This change adds detailed hardware information to the Server CRD.
Hardware info is extracted by the agent from SMBIOS.
The ServerClass CRD is also updated so more precise qualifiers can be used.

Co-authored-by: Gerard de Leeuw <gdeleeuw@leeuwit.nl>
Signed-off-by: Gerard de Leeuw <gdeleeuw@leeuwit.nl>
Signed-off-by: Andrey Smirnov <andrey.smirnov@talos-systems.com>

### CODE REVIEW

File: app/sidero-controller-manager/config/samples/metal_v1alpha2_environment.yaml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+apiVersion: metal.sidero.dev/v1alpha2
+kind: Environment
+metadata:
+  name: default
+spec:
+  kernel:
+    url: "https://github.com/talos-systems/talos/releases/download/v0.10.3/vmlinuz-amd64"
```
Comment (frezbo) [inline app/sidero-controller-manager/config/samples/metal_v1alpha2_environment.yaml:7]: probably just use latest

File: templates/cluster-template-talos-v0.13.4.yaml [hunk 2]
```diff
@@ -38,7 +38,7 @@ spec:
   template:
     spec:
       serverClassRef:
-        apiVersion: metal.sidero.dev/v1alpha1
+        apiVersion: metal.sidero.dev/v1alpha2
```
Comment (frezbo) [inline templates/cluster-template-talos-v0.13.4.yaml:41]: should be bump to v1alpha2 for the talos v0.13.4 template?

**Review by frezbo — COMMENTED** [review]
release note update about the spec change is v1alpha2

File: app/sidero-controller-manager/config/samples/metal_v1alpha2_environment.yaml (hunk 1 above)
Comment (smira) [reply app/sidero-controller-manager/config/samples/metal_v1alpha2_environment.yaml:7]: yep, good point, and siderolabs, I wonder if someone is actually using those samples :shrug:

File: templates/cluster-template-talos-v0.13.4.yaml (hunk 2 above)
Comment (smira) [reply templates/cluster-template-talos-v0.13.4.yaml:41]: yep, as Sidero still supports 0.13 (we are testing with it), so we should keep the template up to date with the resources

**Review by Unix4ever — APPROVED** [review]

**Review by frezbo — APPROVED** [review]

**Pull request merged**